Backpack/Button/Classes/BPKButton.m

```diff
@@ -0,0 +1,308 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKButton.h"
+#import <Backpack/Color.h>
+#import <Backpack/BPKGradientView.h>
+#import <Backpack/BPKGradient.h>
+
+NS_ASSUME_NONNULL_BEGIN
+@interface BPKButton()
+
+@end
+
+@implementation BPKButton {
+    BPKGradientView *_backgroundView;
+}
+
+- (instancetype)initWithSize:(BPKButtonSize)size style:(BPKButtonStyle)style {
+    self = [super initWithFrame:CGRectZero];
+
+    if (self) {
+        [self setupWithSize:size
+                      style:style
+              imagePosition:BPKButtonImagePositionRight];
+    }
+
+    return self;
+}
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+
+    if (self) {
+        [self setupWithSize:BPKButtonSizeDefault
+                      style:BPKButtonStylePrimary
+              imagePosition:BPKButtonImagePositionRight];
+    }
+
+    return self;
+}
+
+- (void)setupWithSize:(BPKButtonSize)size style:(BPKButtonStyle)style imagePosition:(BPKButtonImagePosition)imagePosition {
+    [self setClipsToBounds:YES];
+    [self setAdjustsImageWhenHighlighted:NO];
+    [self setAdjustsImageWhenDisabled:NO];
+    [self.titleLabel setTextAlignment:NSTextAlignmentCenter];
+    
+    _backgroundView = [[BPKGradientView alloc] initWithGradient:[self gradientWithTopColor:BPKColor.clear bottomColor:BPKColor.clear]];
+    [_backgroundView setUserInteractionEnabled:NO];
+    [self insertSubview:_backgroundView atIndex:0];
```

Comment:
Hmm adding subViews directly to `UIButton` makes me a bit scared. Maybe we should use the gradient layer here instead